### PR TITLE
Support generating nodeedits when form is not present

### PR DESCRIPTION
**Pull request by mikemoritz**
- Adds a new buid:form row in bybuid table so that edits can be generated for props, nodedata, edges when the parent ndef does not exist in the layer
- Add form is None check into setters
- Generate nodeedits for edges (previously not implemented)
- Fixes a bug where nodes with tags in the parent only raises an error when attempting to delete in the child

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -1912,10 +1912,10 @@ async def iterLayerNodeEdits(self):
                 nodeedits[2].append(edit)
                 continue
 
-            if flag == 1:
-                if not nodeedits[0] == buid:
-                    continue
+            if not nodeedits[0] == buid:
+                nodeedits = (buid, None, [])
```
Comment (invisig0th) [inline synapse/lib/layer.py:1916]: WIth a None form, i'm not sure the node edit will be consumable correct?

Comment (mikemoritz) [reply synapse/lib/layer.py]: Updated with form lookups and form None check in node set / tag set editors

File: synapse/lib/layer.py [hunk 2]
```diff
@@ -1076,6 +1076,24 @@ def getAbrvProp(self, abrv):
 
         return s_msgpack.un(byts)
 
+    async def getFormInByProp(self, buid):
```
Comment (invisig0th) [inline synapse/lib/layer.py:1079]: Is this method going to be used exclusively internally? If so, we should implement with a _ so epiphyte doesn't plumb it through the cluster layer :)

File: synapse/lib/layer.py [hunk 3]
```diff
@@ -1076,6 +1076,24 @@ def getAbrvProp(self, abrv):
...
+        '''
+        Retrieve form name for a buid using prop index.
+        '''
+        for lkey, lval in self.layrslab.scanByFull(db=self.byprop):
```
Comment (invisig0th) [inline synapse/lib/layer.py:1083]: omg this will do a full table scan to reach one where lval == buid :)

File: synapse/lib/layer.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/layer.py:1079]: If its only ever called by the *layer* itself I shouldn't need to touch it. The real problems arise from *external* callers to layer APIs.

File: synapse/lib/layer.py [hunk 4]
```diff
@@ -1120,6 +1120,17 @@ def getAbrvProp(self, abrv):
 
         return s_msgpack.un(byts)
 
+    async def _getFormByBuid(self, buid):
```
Comment (vEpiphyte) [inline synapse/lib/layer.py:1123]: This is vestigial, can it be removed?

**Pull request merged**